src/util.cpp

```diff
@@ -545,66 +545,46 @@ fs::path AbsPathForConfigVal(const fs::path& path, bool net_specific)
 
 fs::path GetDefaultDataDir()
 {
-    // Windows < Vista: C:\Documents and Settings\Username\Application Data\gridcoinresearch
-    // Windows >= Vista: C:\Users\Username\AppData\Roaming\gridcoinresearch
-    // Mac: ~/Library/Application Support/gridcoinresearch
-    // Unix: ~/.gridcoin
+    // Windows < Vista: C:\Documents and Settings\Username\Application Data\GridcoinResearch
+    // Windows >= Vista: C:\Users\Username\AppData\Roaming\GridcoinResearch
+    // Mac: ~/Library/Application Support/GridcoinResearch
+    // Linux/Unix: ~/.GridcoinResearch
 #ifdef WIN32
     // Windows
+
+    // This is the user's base roaming AppData path with GridcoinResearch added.
     return GetSpecialFolderPath(CSIDL_APPDATA) / "GridcoinResearch";
 #else
-        //2-25-2015
-        fs::path pathRet;
-        char* pszHome = getenv("HOME");
+    fs::path pathRet;
 
-        if (mapArgs.count("-datadir"))
-        {
-            fs::path path2015 = fs::system_complete(mapArgs["-datadir"]);
-            if (fs::is_directory(path2015))
-            {
-                pathRet = path2015;
-            }
-        }
-        else
-        {
-            if (pszHome == NULL || strlen(pszHome) == 0)
-                pathRet = fs::path("/");
-            else
-                pathRet = fs::path(pszHome);
-        }
-#ifdef MAC_OSX
-    // Mac
-    pathRet /= "Library/Application Support";
-    fs::create_directory(pathRet);
+    // For everything except for Windows, use the environment variable to get
+    // the home path.
+    char* pszHome = getenv("HOME");
 
-    if (mapArgs.count("-datadir"))
-    {
-        return pathRet;
-    }
-    else
-    {
-        return pathRet / "GridcoinResearch";
+    // There is no home path, so default to the root directory.
+    if (pszHome == NULL || strlen(pszHome) == 0) {
```

Comment:
Can `NULL` here be replaced with `nullptr`?

Reply:
Yes. NULL is still valid, but we should modernize the usage.